h2o-genmodel/src/main/java/hex/genmodel/algos/targetencoder/TargetEncoderMojoModel.java

```diff
@@ -0,0 +1,77 @@
+package hex.genmodel.algos.targetencoder;
+
+import hex.genmodel.MojoModel;
+import hex.genmodel.easy.RowData;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class TargetEncoderMojoModel extends MojoModel implements FeatureTransformerModel{
+
+  public TargetEncoderMojoModel(String[] columns, String[][] domains, String responseName) {
+    super(columns, domains, responseName);
+  }
+
+  public EncodingMaps _targetEncodingMap;
+  public boolean _withBlending;
+  public double _inflectionPoint;
+  public double _smoothing;
+
+  // Could be passed from the model training phase so that we don't have to recompute it here. Or maybe it is fine as it should be computed only once per mojoModel
+  private double _priorMean = -1; 
+
+  @Override
+  public RowData transform(RowData data) {
+    
+    if(_targetEncodingMap != null) {
+      for (Map.Entry<String, EncodingMap> columnToEncodingsMap : _targetEncodingMap.entrySet()) {
+        String columnName = columnToEncodingsMap.getKey();
+        String originalValue = (String) data.get(columnName);
+        EncodingMap encodings = columnToEncodingsMap.getValue();
+
+        _priorMean = _priorMean == -1 ? computePriorMean(encodings) : _priorMean;
+        
+        int[] correspondingNumAndDen = encodings._encodingMap.get(originalValue);
+
+        int numberOfRowsInCurrentCategory = correspondingNumAndDen[1];
+        double lambda = computeLambda(numberOfRowsInCurrentCategory, _inflectionPoint, _smoothing);
+        double posteriorMean = (double) correspondingNumAndDen[0] / correspondingNumAndDen[1];
+        double blendedValue = computeBlendedEncoding(lambda, posteriorMean, _priorMean);
```

Comment:
Would it be possible to calculate `lambda`, `priorMean`, `posteriorMean` and `blendedValue` during the training phase? If so, we could improve on performance here.

Reply:
@mn-mikke It is possible and this is stated few lines higher (line 20) in my comment. This is an optimisation that can be definitely done later. Ultimately we don't need anything but Map<category, final_encoding>, where `final_encoding` might be with or without blending... so we actually can get rid of all blending hyper parameters and don't sent int[] with numerator and denominator separately. We can precompute map with final encodings when we are storing model as Mojo. **But all this is true only under assumption that we don't give an ability for users to change hyperparameters at PROD prediction stage( ie we are using same parameters as during training).** 
In my current version I assumed that we can decide whether to use blending for PROD predictions or not and even play with hps.... see `TargetEncoderMojoModelTest.transformWithoutBlending`. That is my comment about pre-computation was only at `priorMean` as it is used in all cases. We have many options... we can make it not allowed to change blending hps and send two pre-computed final encodings : `with_blending` and `without_blending`.